rust/storage/src/admissioncontrolleds3.rs

```diff
@@ -50,12 +50,37 @@ pub struct AdmissionControlledS3Storage {
     rate_limiter: Arc<RateLimitPolicy>,
 }
 
+#[derive(Debug, Clone, Copy)]
+pub enum StorageRequestPriority {
+    High = 0,
+    Medium = 1,
+}
```

Comment:
wondering if we want to explicitly tag these with "foreground" and "background" or if we anticipate needing more priority classes in the future

Reply:
I would prefer if we just didn't use semantic names and instead numeric values. The names always become meaningless or overconstrained in terms of what they represent. 0->N is easier and more flexible.

Reply:
Yeah I had the same thought this morning. Made it P0, P1, P2...